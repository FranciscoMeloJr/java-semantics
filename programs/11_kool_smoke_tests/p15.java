class c1 {
  int v;
  void c1() { v=7;++v; }
  int m0()  { return v; }
  int m1()  { return -v; }
}

class c2 {
  int v;
  void c2(int x) { v=x; }
  int m0()  { return v; }
  int m1()  { return -v; }
}

class main {
  void main(String[] args) {
    c1 o1 = new c1();
    c2 o2 = new c2(1);
    System.out.print(o1.m0());
    o1.m0 = o1.m1;
    System.out.print(o1.m0());
    o1.m0 = o2.m0;
    System.out.print(o1.m0());
    o1.m0 = o2.m1;
    System.out.print(o1.m0());
    o2.c2(3);
    System.out.print(o1.m0());
  }
}

