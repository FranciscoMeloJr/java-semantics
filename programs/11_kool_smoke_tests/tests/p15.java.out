8-81-1-3
Done!
